jmeter-java-dsl/src/main/java/us/abstracta/jmeter/javadsl/codegeneration/params/ChildrenParam.java

```diff
@@ -121,12 +121,16 @@ private static int findExecutionOrder(Class<?> returnType) {
   }
 
   public void addChild(MethodCall child) {
+    validateChildType(child);
+    children.add(child);
+  }
+
+  private void validateChildType(MethodCall child) {
```

Comment:
I think the logic is simple enough (and not reused) to keep in the original method.